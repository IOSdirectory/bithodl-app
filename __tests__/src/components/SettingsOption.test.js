import React from 'react';
import renderer from 'react-test-renderer';
import SettingsOption from '../../../src/components/SettingsOption';

describe('SettingsOption', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <SettingsOption
        name='876f5e88-32ad-444c-8b1a-2d06220dd517'
        value='b7c127e4-0058-4349-af1f-216202322326'
        onSelect={() => {}}
      />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly as the last item', () => {
    const tree = renderer.create(
      <SettingsOption
        name='f8cf85fb-3834-47e6-a8b8-a669e78df02f'
        value='b0a6df8b-db83-4386-a9b8-0f09f252c672'
        onSelect={() => {}}
        isLastItem={true}
      />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
