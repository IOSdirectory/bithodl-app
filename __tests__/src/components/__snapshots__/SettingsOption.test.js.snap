// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SettingsOption renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
      },
      undefined,
    ]
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Array [
        Object {
          "borderBottomColor": "#DDDDDF",
          "borderBottomWidth": 0.5,
          "marginLeft": 15,
          "paddingBottom": 12,
          "paddingRight": 15,
          "paddingTop": 12,
        },
        undefined,
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      numberOfLines={undefined}
      style={
        Array [
          Object {
            "color": "#504F63",
            "fontFamily": "Avenir Next",
            "fontWeight": "400",
          },
          Object {
            "fontWeight": "500",
          },
        ]
      }
    >
      876f5e88-32ad-444c-8b1a-2d06220dd517
    </Text>
  </View>
</View>
`;

exports[`SettingsOption renders correctly as the last item 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  hasTVPreferredFocus={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
      },
      undefined,
    ]
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Array [
        Object {
          "borderBottomColor": "#DDDDDF",
          "borderBottomWidth": 0.5,
          "marginLeft": 15,
          "paddingBottom": 12,
          "paddingRight": 15,
          "paddingTop": 12,
        },
        Object {
          "borderBottomWidth": 0,
        },
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      numberOfLines={undefined}
      style={
        Array [
          Object {
            "color": "#504F63",
            "fontFamily": "Avenir Next",
            "fontWeight": "400",
          },
          Object {
            "fontWeight": "500",
          },
        ]
      }
    >
      f8cf85fb-3834-47e6-a8b8-a669e78df02f
    </Text>
  </View>
</View>
`;
